core/src/main/java/com/oath/oak/OakRKeyBufferImpl.java

```diff
@@ -68,4 +68,9 @@ public double getDouble(int index) {
         return transformer.apply(byteBuffer);
     }
 
+    @Override
+    public void unsafeBufferToIntArrayCopy(int srcPosition, int[] dstArray, int countInts) {
+        UnsafeUtils.unsafeBufferToArrayCopy(byteBuffer, srcPosition, dstArray, countInts);
```

Comment:
What if keys buffer is being copied during rebalance? Is it that source key buffer is immutable anyway? Keys are anyway immutable in current usage... Good to add a comment about that, to show that this was considered

Reply:
keys are immutable and the current mechanism for releasing a key is only when no references so no problem